zeppelin-plugins/launcher/docker/src/main/java/org/apache/zeppelin/interpreter/launcher/DockerInterpreterProcess.java

```diff
@@ -242,18 +233,22 @@ public void progress(ProgressMessage message) throws DockerException {
       throw new IOException(e.getMessage());
     } catch (InterruptedException e) {
       LOGGER.error(e.getMessage(), e);
+      // Restore interrupted state...
+      Thread.currentThread().interrupt();
       throw new IOException(e.getMessage());
     }
 
     long startTime = System.currentTimeMillis();
 
     // wait until interpreter send dockerStarted message through thrift rpc
     synchronized (dockerStarted) {
-      if (!dockerStarted.get()) {
+      while (!dockerStarted.get()) {
```

Comment:
It sometimes can make an infinite loop even if docker is really stable and guarantee to return in time. How about maximum retry or timeout like the below code? WDYT?

Reply:
Your right, I will change this.

Reply:
I have adapted the corresponding passage. Please look over it again.